scipy/optimize/_constraints.py

```diff
@@ -192,6 +202,30 @@ def __repr__(self):
             end = ")"
         return start + end
 
+    def residual(self, x):
```

Comment:
`PreparedConstraint` has a `violation` method, which is very similar to this. Is there any rationalisation possible:

- should we rename `violation-->residual`?
- (others)?

Also, can we add something to the effect of "if sl, sb are negative then the values in x lie outside the bounds, if they are positive the values in x lie within the bounds"

Reply:
> should we rename `violation-->residual`

Fine with me. It's not public, so it can be changed. But the definition is the opposite of `residual`. I chose `residual` because that is what we ended up with as the [standard name](https://docs.scipy.org/doc/scipy/reference/optimize.linprog-highs-ds.html#optimize-linprog-highs-ds) for the distance to a constraint (bound, inequality, and equality constraint) `linprog`'s result object.

> Also, can we add something to the effect of

Yes, I was thinking about doing that. Will do. I wanted to phrase it in a way that is precise for arrays. Also need to be careful because when an element of `x` is out of bounds, only one of the corresponding elements in `sb` and `sl` will be negative, not both.

```
        When all elements of ``sl`` and ``sb`` are positive, all elements of
        ``x`` lie within the bounds; a negative element in ``sl`` or ``sb``
        indicates that the corresponding element of ``x`` is out of bounds.
```